### DISCUSSION TITLE
oauth/connect no longer working

### DISCUSSION THREAD

**Discussion by elcombe** [etsy/open-api#689] [Q&A] [answered]
We recently upgraded to the new Etsy V3 and moved to OAuth 2.0.

A week ago or so it was working, then with no code or Etsy configuration changes am seeing the following

calling `https://www.etsy.com/oauth/connect?client_id=791ppa8xr5byamf8v3iqxebg&code_challenge=-3YUZwAJrU7LFZ9W96IjE6GuAdJmkShBPJRf3mugh-k&code_challenge_method=S256&redirect_uri=https%3A%2F%2Fstaging.chitchats.com%2Fauth%2Fetsy%2Fcallback&response_type=code&scope=transactions_r+transactions_w+listings_r&state=0e5aa127b854dd72d4ef8236550580cc23991bddc490c7cd`

now returns the following error:

`CORS error invalid redirect`,

I suspect there's something wrong with the callback urls, but they seem to match

<img width="754" alt="Screenshot 2022-12-07 at 4 16 51 PM" src="https://user-images.githubusercontent.com/22106637/206298200-850636a2-7b87-4785-bd4f-48af801949c7.png">

I've also tried hard coding the redirect_uri, and the only thing that seems to make a difference is adding www. to both the redirect_uri and the callback url list.

What is misconfigured that is disallowing my oauth/connect call and did anything change over the past week or so that could have caused this issue? 

Thanks

**Comment by etsyachristensen** [comment, marked as answer]
It would appear that you may need to enabled CORS on your staging server for *.chitcats.com. This isn't something on our end as far as we can tell. That error doesn't look like something our server would return.

**Reply by elcombe** [reply]
@etsyachristensen We'll test that shortly, but it seems odd to us that the CORS would be to allow ourselves in. For a bit more context here's the full CORS error we were able to see when calling Etsy oauth/connect: "cross origin resource sharing error preflight disallowed redirect". It used to show a message about allowing consent for the app to access the specified roles in Etsy, but now it errors before that. If we log out of Etsy it will ask us to login to etsy, but errors after login.

**Reply by elcombe** [reply]
Following up after testing, we set CORS to allow * on the staging server and tested again and were met with the same error.

**Reply by etsyachristensen** [reply]
Did you change anything about how your code does oauth since it was working? Since you are on a staging server here, I assume you're working on some new code, but does it in anyway connect with the oauth code?

Also, are you certain you have the SSL enabled on your staging server domain? Sometimes you'll get a redirect issue if the SSL isn't turned on.

Beyond that, I'm not really sure how to help. We haven't changed anything on our side in weeks when it comes to oauth.

**Reply by etsyachristensen** [reply]
@elcombe I've discussed with one of our network engineers.

> CORS errors only get thrown when there's a cross-domain javascript request. The request to /oauth/connect must be a direct redirect of the user, not loaded by javascript. So either the developer is trying to hit /oauth/connect from javascript (which intentionally fails with a CORS error, as that would be a serious security issue), or the error is coming from a different cross-origin request that we're not involved with.
> 
> The other possibility is that this error is being shown on [etsy.com](http://etsy.com/) during the signin process. If that's the case, we'll need more info about what cross-origin request failed (should be visible in the network tab of the chrome/firefox inspector).

**Reply by elcombe** [reply]
@etsyachristensen We've been able to locate this is an issue on our side. Will close this discussion. Thanks!

**Reply by etsyachristensen** [reply]
Ok thanks!
